numaprom/clients/prometheus.py

```diff
@@ -29,32 +29,36 @@ def query_metric(
         if end < start:
             raise ValueError("end_time must not be before start_time")
 
-        result = self.query_range(query, start, end, step)
-
-        arr = np.array(result["values"])
-        df = pd.DataFrame(arr, columns=["timestamp", metric_name])
-        df = df.astype(float)
-
-        data = result["metric"]
-        if return_labels:
-            for label in return_labels:
-                if label in data:
-                    df[label] = data[label]
-
-        if not df.empty:
-            df.set_index("timestamp", inplace=True)
-            df.index = pd.to_datetime(df.index.astype(int), unit="s")
+        results = self.query_range(query, start, end, step)
+
+        frames = []
+        for result in results:
+            arr = np.array(result["values"])
+            _df = pd.DataFrame(arr, columns=["timestamp", metric_name])
+            _df = _df.astype(float)
+
+            data = result["metric"]
+            if return_labels:
+                for label in return_labels:
+                    if label in data:
+                        _df[label] = data[label]
+            frames.append(_df)
+
+        df = pd.concat(frames, ignore_index=True)
+        df.sort_values(by=["timestamp"], inplace=True)
+        df["timestamp"] = pd.to_datetime(df["timestamp"], unit="s")
+        df.reset_index(drop=True, inplace=True)
```

Comment:
Can we return the df with timestamp as the index? Or will that break things?

Reply:
No, timestamps will have duplicates (from multiple hashes). But `clean_data` method handles that.